libtiledbvcf/src/dataset/variant_stats.cc

```diff
@@ -0,0 +1,325 @@
+/**
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2022 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "dataset/variant_stats.h"
+#include "utils/logger_public.h"
+#include "utils/utils.h"
+
+namespace tiledb::vcf {
+
+//===================================================================
+//= public static functions
+//===================================================================
+
+void VariantStats::create(
+    Context& ctx, std::string root_uri, tiledb_filter_type_t checksum) {
+  LOG_DEBUG("VariantStats: Create array");
+
+  // Create filters
+  Filter zstd(ctx, TILEDB_FILTER_ZSTD);
+
+  FilterList filters_1(ctx);
+  FilterList filters_2(ctx);
+  filters_1.add_filter(zstd);
+  filters_2.add_filter({ctx, TILEDB_FILTER_DOUBLE_DELTA});
```

Comment:
For offsets I think we should also add bitwidth reduction

Reply:
Added.